### PR TITLE
Removes Projectile Megafauna Cheesing

### PR DESCRIPTION
**Pull request by Whoneedspacee**
## About The Pull Request

Dead mobs are now dusted by colossus typed projectiles.

Colossus typed projectiles are always considered a direct target projectile now so you cannot lay down and ignore them anymore.

Wendigo and frost miner projectiles are now subtypes of this projectile type as well.

## Why It's Good For The Game

Defeating megafauna with corpses in chairs and laying on the ground is kinda stupid.

## Changelog
:cl:
fix: Megafauna are no longer stopped by corpses and chairs.
/:cl:

### CODE REVIEW

**Comment by Whoneedspacee** [comment]
megafauna cheesers in shambles

File: code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm [hunk 1]
```diff
@@ -255,17 +255,29 @@
...
 	speed = 2
 	eyeblur = 0
 	damage_type = BRUTE
 	pass_flags = PASSTABLE
+	var/explode_hit_objects = TRUE
+
+/obj/projectile/colossus/can_hit_target(atom/target, direct_target = FALSE, ignore_loc = FALSE, cross_failed = FALSE)
+	if(isliving(target))
+		direct_target = TRUE
+	. = ..(target, direct_target, ignore_loc, cross_failed)
```
Comment (Fikou) [inline code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm:271]: return ..() at the end of procs

**Comment by tralezab** [comment]
excellent fix my friend

File: code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm [hunk 2]
```diff
@@ -255,17 +255,29 @@
...
+	return ..(target, direct_target, ignore_loc, cross_failed)
 
 /obj/projectile/colossus/on_hit(atom/target, blocked = FALSE)
 	. = ..()
+	if(isliving(target))
+		var/mob/living/dust_mob = target
+		if(dust_mob.stat == DEAD)
+			dust_mob.dust()
+	if(!explode_hit_objects)
+		return
```
Comment (Rohesie) [inline code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm:280]:
```suggestion
	if(isliving(target))
		var/mob/living/dust_mob = target
		if(dust_mob.stat == DEAD)
			dust_mob.dust()
		return
	if(!explode_hit_objects)
		return
```

**Pull request merged**